File: Week6/problems/sentimental-mario-more/mario.py
# TODO
from cs50 import get_int


def main():
    height = get_height()

    # Build the first half
    for i in range(height):
        for j in range(height - i - 1):
            print(" ", end="")

        for k in range(i + 1):
            print("#", end="")

        # Build the second half
        print("  ", end="")
        for k in range(i + 1):
            print("#", end="")

        print("")


# Get user input for height
def get_height():
    while True:
        try:
            n = get_int("Height: ")
            if (n > 0) & (n < 9):
                return n
        except ValueError:
            print("Not an integer")


main()

File: Week1/problems/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Get user input
    int height;
    do
    {
        height = get_int("Height: ");
    }
    // Check if height is between 0 and 9
    while (height < 1 || height > 8);

    // Print 'height' number of rows
    for (int row = 1; row <= height; row++)
    {
        // Number of spaces per row
        for (int w = 0; w < height - row; w++)
        {
            printf(" ");
        }

        // Number of '#'s per row
        for (int x = 0; x < row; x++)
        {
            printf("#");
        }

        // Print space inbetween
        printf("  ");

        // Print second half

        for (int y = 0; y < row; y++)
        {
            printf("#");
        }

        // Create new line
        printf("\n");
    }

}